scipy/fft/_basic_backend.py

```diff
@@ -0,0 +1,183 @@
+        raise ValueError(arg_err_msg("plan"))
+    if norm is None:
+        norm = 'backward'
+    return norm
+
+
+def _execute(func_str, pocketfft_func, x, s, axes, norm, 
+             overwrite_x, workers, plan):
+    xp = array_namespace(x)
+    # pocketfft is used whenever SCIPY_ARRAY_API is not set,
+    # or x is a NumPy array or array-like.
+    # When SCIPY_ARRAY_API is set, we try to use xp.fft for CuPy arrays,
+    # PyTorch arrays and other array API standard supporting objects.
+    # If that is unsuccessful, we attempt to convert to np and back to use pocketfft.
+    if is_numpy(xp):
+        return pocketfft_func(x, s, axes, norm=norm, overwrite_x=overwrite_x,
+                              workers=workers, plan=plan)
+
+    norm = _validate_fft_args(overwrite_x, workers, plan, norm)
+    if hasattr(xp, 'fft'):
+        xp_func = getattr(xp.fft, func_str)
+        return xp_func(x, s, axes, norm=norm)
+
+    x = np.asarray(x)
+    y = pocketfft_func(x, s, axes, norm=norm)
+    return xp.asarray(y)
+
+
+def fft(x, n=None, axis=-1, norm=None,
+        overwrite_x=False, workers=None, *, plan=None):
+    return _execute('fft', _pocketfft.fft, x, n, axis, norm,
+                    overwrite_x, workers, plan)
+
+
+def ifft(x, n=None, axis=-1, norm=None, overwrite_x=False, workers=None, *,
+         plan=None):
+    return _execute('ifft', _pocketfft.ifft, x, n, axis, norm,
+                    overwrite_x, workers, plan)
+
+
+def rfft(x, n=None, axis=-1, norm=None,
+         overwrite_x=False, workers=None, *, plan=None):
+    return _execute('rfft', _pocketfft.rfft, x, n, axis, norm,
+                    overwrite_x, workers, plan)
+
+
+def irfft(x, n=None, axis=-1, norm=None,
+          overwrite_x=False, workers=None, *, plan=None):
+    return _execute('irfft', _pocketfft.irfft, x, n, axis, norm,
+                    overwrite_x, workers, plan)
+
+
+def hfft(x, n=None, axis=-1, norm=None,
+         overwrite_x=False, workers=None, *, plan=None):
+    return _execute('hfft', _pocketfft.hfft, x, n, axis, norm,
+                    overwrite_x, workers, plan)
+
+
+def ihfft(x, n=None, axis=-1, norm=None,
+          overwrite_x=False, workers=None, *, plan=None):
+    return _execute('ihfft', _pocketfft.ihfft, x, n, axis, norm,
+                    overwrite_x, workers, plan)
+
+
+def fftn(x, s=None, axes=None, norm=None,
+         overwrite_x=False, workers=None, *, plan=None):
+    return _execute('fftn', _pocketfft.fftn, x, s, axes, norm,
+                    overwrite_x, workers, plan)
+
+
+
+def ifftn(x, s=None, axes=None, norm=None,
+          overwrite_x=False, workers=None, *, plan=None):
+    return _execute('ifftn', _pocketfft.ifftn, x, s, axes, norm,
+                    overwrite_x, workers, plan)
+
+
+def fft2(x, s=None, axes=(-2, -1), norm=None,
+         overwrite_x=False, workers=None, *, plan=None):
+    xp = array_namespace(x)
+    x = np.asarray(x)
+    y = _pocketfft.fft2(x, s=s, axes=axes, norm=norm,
```

Comment:
Why don't we just move the `fft2` implementation here. It just calls `fftn`. 

Same applies to all other `*fft2` functions.

Reply:
This is planned for a follow-up to come soon.

Reply:
It may be useful to open this follow-up issue and add the planned items - then we can resolve the comments here.

Reply:
Issue is open gh-19257.